source/python/idetect/model.py

```diff
@@ -63,8 +86,95 @@ def update_status(self, new_status):
             })
         if result != 1:
             try:
-                updated = session.query(Article).filter(Article.id == self.id).one()
-                raise UnexpectedArticleStatusException(self, expected_status, updated.status)
+                updated = session.query(Article).filter(
+                    Article.id == self.id).one()
+                raise UnexpectedArticleStatusException(
+                    self, expected_status, updated.status)
             except NoResultFound:
-                raise UnexpectedArticleStatusException(self, expected_status, None)
+                raise UnexpectedArticleStatusException(
+                    self, expected_status, None)
+
+
+class Content(Base):
+    __tablename__ = 'content'
+
+    id = Column(Integer, primary_key=True)
+    article = relationship(
+        'Article', secondary=article_content, back_populates='content')
+    content = Column(String)
+    content_type = Column(String)
+
+
+class ReportUnit:
+    PEOPLE = 'people'
+    HOUSEHOLDS = 'households'
+
+
+class ReportTerm:
+    DISPLACED = 'displaced'
+    EVACUATED = 'evacuated'
+    FLED = 'forced to flee'
+    HOMELESS = 'homeless'
+    CAMP = 'in relief camp'
+    SHELTERED = 'sheltered'
+    RELOCATED = 'relocated'
+    DESTROYED = 'destroyed housing'
+    DAMAGED = 'partially destroyed housing'
+    UNINHABITABLE = 'uninhabitable housing'
+
+
+report_location = Table(
+    'report_location', Base.metadata,
+    Column('report', ForeignKey('report.id'), primary_key=True),
+    Column('location', ForeignKey('location.id'), primary_key=True)
+)
 
+
+class Report(Base):
+    __tablename__ = 'report'
+
+    id = Column(Integer, primary_key=True)
+    article_id = Column('article', Integer, ForeignKey(
+        'article.id'), primary_key=True)
+    article = relationship('Article', back_populates='reports')
+    sentence_start = Column(Integer)
+    sentence_end = Column(Integer)
+    reporting_unit = Column(String)
+    reporting_term = Column(String)
+    mentions_displacement_figure = Column(Boolean)
```

Comment:
In general, I'd err on the side of not storing derived data unless it's particularly hard to recalculate. That way there's never a question of whether the derived data is up to date.